pkg/grpc/tls_client_certificate_authenticator_test.go

```diff
@@ -99,16 +104,21 @@ func TestTLSClientCertificateAuthenticator(t *testing.T) {
 	expectedMetadata := auth.MustNewAuthenticationMetadataFromProto(&auth_pb.AuthenticationMetadata{
 		Public: structpb.NewStructValue(&structpb.Struct{
 			Fields: map[string]*structpb.Value{
-				"username": structpb.NewStringValue("John Doe"),
+				"email": structpb.NewStringValue("me@example.com"),
 			},
 		}),
 	})
-	authenticator := bb_grpc.NewTLSClientCertificateAuthenticator(clientCAs, clock, expectedMetadata)
+
+	aValidator := jmespath.MustCompile(`contains(dnsNames, 'a.example.com')`)
+	bValidator := jmespath.MustCompile(`contains(dnsNames, 'b.example.com')`)
+	metadataExtractor := jmespath.MustCompile(`{"public": {"email": emailAddresses[?ends_with(@, '@example.com')] | [0] } }`)
```

Comment:
My recommendation would be to use a simpler expression here like `{"public": @}`, because that would allow us to validate the format of everything that comes out of this.